### ISSUE TITLE
oci_identity_api_key fails with AttributeError: 'bool' object has no attribute 'split' when user has API key

### ISSUE THREAD

**Issue by krilor** [oracle/oci-ansible-collection#170]
## Issue Report

**Describe the issue**

oci_identity_api_key fails if user already has a key. If the user does not have a key, everything works.

```
runpy.run_module(mod_name='ansible_collections.oracle.oci.plugins.modules.oci_identity_api_key', init_globals=dict(_module_fqn='ansible_collections.oracle.oci.plugins.modules.oci_identity_api_key', _modlib_path=modlib_path),
  File "/usr/lib/python3.8/runpy.py", line 207, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/modules/oci_identity_api_key.py", line 298, in <module>
  File "/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/modules/oci_identity_api_key.py", line 292, in main
  File "/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_resource_utils.py", line 789, in create
  File "/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_resource_utils.py", line 744, in get_matching_resource
  File "/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_common_utils.py", line 252, in compare_dicts
  File "/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_common_utils.py", line 833, in is_any_key_matching_stack
  File "/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_common_utils.py", line 841, in is_key_matching_stack
AttributeError: 'bool' object has no attribute 'split'
fatal: [10.141.69.60 -> localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/kristoffer.lorentsen/.ansible/tmp/ansible-tmp-1661881630.239003-6466-122142134784610/AnsiballZ_oci_identity_api_key.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/kristoffer.lorentsen/.ansible/tmp/ansible-tmp-1661881630.239003-6466-122142134784610/AnsiballZ_oci_identity_api_key.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/kristoffer.lorentsen/.ansible/tmp/ansible-tmp-1661881630.239003-6466-122142134784610/AnsiballZ_oci_identity_api_key.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.oracle.oci.plugins.modules.oci_identity_api_key', init_globals=dict(_module_fqn='ansible_collections.oracle.oci.plugins.modules.oci_identity_api_key', _modlib_path=modlib_path),\n  File \"/usr/lib/python3.8/runpy.py\", line 207, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.8/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/modules/oci_identity_api_key.py\", line 298, in <module>\n  File \"/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/modules/oci_identity_api_key.py\", line 292, in main\n  File \"/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_resource_utils.py\", line 789, in create\n  File \"/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_resource_utils.py\", line 744, in get_matching_resource\n  File \"/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_common_utils.py\", line 252, in compare_dicts\n  File \"/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_common_utils.py\", line 833, in is_any_key_matching_stack\n  File \"/tmp/ansible_oracle.oci.oci_identity_api_key_payload_u85rl6wv/ansible_oracle.oci.oci_identity_api_key_payload.zip/ansible_collections/oracle/oci/plugins/module_utils/oci_common_utils.py\", line 841, in is_key_matching_stack\nAttributeError: 'bool' object has no attribute 'split'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

Looks like something broke on version 2.46 of the collection and forward. Version 2.45 works as expected with no error.

**Expected behavior**

No error.

**Environment**

* OS version: Ubuntu 20.04

* Ansible version:

```
ansible [core 2.13.3]
  config file = /home/kristoffer.lorentsen/elt/cumulus/ansible-vault/ansible.cfg
  configured module search path = ['/home/kristoffer.lorentsen/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/kristoffer.lorentsen/.local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/kristoffer.lorentsen/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/kristoffer.lorentsen/.local/bin/ansible
  python version = 3.8.10 (default, Jun 22 2022, 20:18:18) [GCC 9.4.0]
  jinja version = 3.0.1
  libyaml = True
```

* OCI Python SDK version:
```
2.80.1
```

* OCI Ansible Collection Version

I've installed different version and it seems it broke with version 2.46

```
ansible-galaxy collection install -f oracle.oci:2.45.0 <- works
ansible-galaxy collection install -f oracle.oci:2.46.0 <- broken
ansible-galaxy collection install -f oracle.oci <- broken
```

**Ansible playbook to reproduce the issue**

```
---
- name: repro api key error
  hosts:
    - localhost
  become: false
  gather_facts: false

  tasks:
    - name: add key
      oracle.oci.oci_identity_api_key:
        user_id: "ocid1.user.oc1..a.."
        key: |-
          -----BEGIN PUBLIC KEY-----
          ...
          -----END PUBLIC KEY-----
        state: present
```

**Comment by dineshsuthar786** [comment]
@krilor Thanks for reporting the issue. We are checking on this and will get back to you.

**Comment by dineshsuthar786** [comment]
@krilor we have identified the bug and resolved it in latest release (v3.0.0). Kindly confirm and let us know if you still face the issue.

**Comment by krilor** [comment]
@dineshsuthar786 I've retested and can confirm that it works :) Thanks for fixing this so fast!

This is the third time I've created an issue/request for this collection and I have been impressed by the help I've gotten on all of them. Keep up the good work - it is really appreciated 💯
